### PR TITLE
Tutorial: Testing intro

### PR DESCRIPTION
**Pull request by killercup**

### CODE REVIEW

File: src/tutorial/testing.md [hunk 1]
```diff
@@ -1,5 +1,101 @@
...
+```rust
+#[test]
+fn check_answer_validity() {
+    assert_eq!(answer(), 42);
+}
+```
+
+You can put this snippet of code in pretty much any file
+and `cargo test` will find it
+and run it.
```
Comment (peddermaster2) [inline src/tutorial/testing.md]: and `cargo test` will find ~it~ and run it.

File: src/tutorial/testing.md [hunk 2]
```diff
@@ -1,5 +1,101 @@
...
+if you haven't heard of it.
+
+</aside>
+
+## Automated testing
+
+Now, this is all fine and dandy,
+but doing all of this manually?
+That can take a lot of time.
+I don't know about you,
```
Comment (artem-zinnatullin) [inline src/tutorial/testing.md]: nit: maybe there is a way to rephrase it to not speak from first person tense in the book?

Particularly here it looks like you can just keep:

>Let's talk about how to automate these tests.

This feeling when you write something (book, docs, code) and want to left some personal touch, yea :)

File: src/tutorial/testing.md [hunk 3]
```diff
@@ -1,5 +1,101 @@
...
+#[test]
+fn check_answer_validity() {
+    assert_eq!(answer(), 42);
+}
+```
+
+You can put this snippet of code in pretty much any file
+and `cargo test` will find it
+and run it.
+All the magic is done by the `#[test]` attribute.
```
Comment (artem-zinnatullin) [inline src/tutorial/testing.md]: Would be great to avoid "magic" in official books/docs, there is no magic unless it's unsafe :D

Something like this could work:

>The key here is the `#[test]` attribute, it allows the build system to discover such functions and run them as tests, verifying that they don't panic.

File: src/tutorial/testing.md (hunk 2 above)
Comment (killercup) [reply src/tutorial/testing.md]: Thanks -- I really wasn't sure about this, either. I know I can just omit the whole sentence… but how about this?

> Now, this is all fine and dandy,
> but doing all of this manually?
> That can take a lot of time.
> _At the same time,
> many people have come to enjoy telling computers to do things for them._
> Let's talk about how to automate these tests.

Comment (artem-zinnatullin) [reply src/tutorial/testing.md]: lgtm

**Comment by killercup** [comment]
Well if CLI tests looked a bit more pretty I'd totally put them in a readme and run docmatic on it!
